test/unit-test/dawn/Optimizer/Passes/TestPassComputeStageExtents.cpp

```diff
@@ -0,0 +1,213 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Compiler/DawnCompiler.h"
+#include "dawn/Compiler/Options.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/SIR/SIRSerializer.h"
+#include "test/unit-test/dawn/Optimizer/Passes/TestEnvironment.h"
+#include <fstream>
+#include <gtest/gtest.h>
+#include <streambuf>
+
+using namespace dawn;
+
+namespace {
+
+TEST(ComputeStageExtents, test_stencil_01) {
+  std::string filename = TestEnvironment::path_ + "/compute_extent_test_stencil_01.sir";
+  std::ifstream file(filename);
+  DAWN_ASSERT_MSG((file.good()), std::string("File " + filename + " does not exists").c_str());
+
+  std::string jsonstr((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+
+  std::shared_ptr<SIR> sir = SIRSerializer::deserializeFromString(jsonstr, SIRSerializer::SK_Json);
```

Comment:
should we pack all of this up to the tests into a function? Seems this is repeated in every test?
or should we use the `TEST_P` env?

Reply:
good point, I did created a fixture, can you have a look?